docs/content/dev/notes.md

```diff
@@ -0,0 +1,58 @@
+
+`groupPermissions` and `userPermissions` each hold a list of the appropriate permissions.
+Each permission holds a reference to a note and a user/group and specify what the user/group is allowed to do.
+Each permission is additive, that means a user that has only the right to read a note via a group, but the right to write via a different group or directly for his user, is able to write in the note.
+
+The `viewCount` is a simple counter that holds how often the note in question was requested.
+
+`owner` is the user that created the note or later got ownership of the note. The current owner is able to change the owner of the note to someone else.
+
+The `revisions` hold all revisions of the note. These are the changes to the note content and by which they were performed.
+
+The `authorColors` each specify for the tuple user and note which color should be used to highlight them.
+
+The `historyEntries` hold the history entries this note is referenced in. They are mainly here for the purpose of deleting the history entries on note deletion.
+
+`description`, `tags` and `title` are each information specified in the frontmatter of the note. They are extracted and saved in the database to allow the history page to show them and do a search for tags without having to do a full-text search or having to parse the tags of each note on search.
+While `description` and `tags` are only specified by the frontmatter, the title is
+- the content of the *title* field of the frontmatter of the note
+- **OR** the content of the *title* field in the *opengraph* field of the frontmatter of the note
+- **OR** the first level 1 heading of the note 
+  
+which ever of these is the first to not be unspecified.  
+All mentioned fields are extracted from the note content by the backend on save or update.
+  
+`version` specifies if a note is an old HedgeDoc 1 note, or a new HedgeDoc 2 note. This is mainly used to redirect old notes form <https://md.example.org/noteid> to <https://md.example.org/n/noteid>.
+
+### Conversion of HedgeDoc 1 notes
+
+First we want to define some terms:
+
+- **noteId**: This refers to the auto-generated id for new notes. (https://demo.hedgedoc.org/Q_Iz5T_lQWGYxne0sbMtwg)
+
+- **shortId**: This refers to the auto-generated short id which is used for "published" notes and slide presentation mode. (https://demo.hedgedoc.org/s/61ZHI6HGE)
+
+- **alias**: This refers to user-defined URLs for notes on instances with Free-URL mode enabled. (https://md.kif.rocks/lowercase)
+
+The noteId, shortId and alias of each HedgeDoc 1 note are saved as HedgeDoc 2 aliases and get a newly generated publicId.
```

Comment:
I'd split that for better understanding.

```suggestion
The noteId, shortId and alias of each HedgeDoc 1 note are saved as HedgeDoc 2 aliases. Each note gets a newly generated publicId.
```

Reply:
fixed